### ISSUE TITLE
Server http cookie Max-Age=0 RFC6265; EID3430

### ISSUE THREAD

**Issue by iamtakingiteasy** [netty/netty#10235]
### Expected behavior
When using strict mode of ServerCookieEncoder, Max-Age=0 should not be part of the encoded header value as per https://www.rfc-editor.org/errata/eid3430

### Actual behavior
Max-Age=0 added even in strict mode.

### Steps to reproduce
1. create cookie, set max-age to 0
2. encode it using strict server cookie encoder

### Minimal yet complete reproducer code (or URL to code)

```
Cookie cookie = new DefaultCookie("foo", "bar");
cookie.setMaxAge(0);
ServerCookieEncoder.STRICT.encode(cookie);
```

### Netty version
4.1.49

### JVM version (e.g. `java -version`)
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10)
OpenJDK 64-Bit Server VM (build 11.0.6+10, mixed mode)

### OS version (e.g. `uname -a`)
Linux T480 5.6.4-arch1-1 #1 SMP PREEMPT Mon, 13 Apr 2020 12:21:19 +0000 x86_64 GNU/Linux

**Comment by normanmaurer** [comment]
@slandelle can you have a look ?

**Comment by slandelle** [comment]
Status of errata Errata 3430 is Rejected.

**Comment by slandelle** [comment]
I really wonder about this.
AFAIK, sending a Set-Cookie header with a max-age <=0 or an expires date in the past is a very common way for servers to have the user-agent forcefully revoke cookies, typically on graceful log out.

**Comment by iamtakingiteasy** [comment]
Errata itself is rejected because it proposes to change format from 

> non-zero-digit = [1-9]
> max-age-av = "Max-Age=" non-zero-digit *DIGIT

to 

> max-age-av = "Max-Age=" 1 *DIGIT

Which is not correct, as it makes impossible to send, say Max-Age=5.

But still, underlying RFC states Max-Age as starting from non-zero digit, and note in mentioned errata highlights that, also reminding  to be strict when sending and permissive when receiving.

The fact most user agents (receivers) are permitting zero-value Max-Age and treat it same as Expires set in the past, does not excuse server (sender) operating laxly.

**Comment by iamtakingiteasy** [comment]
Besides, Max-Age=0 even with Expires set in the past can cause issues with at least for some of chromium builds as per old bug https://bugs.chromium.org/p/chromium/issues/detail?id=416800 and this matrix https://inikulin.github.io/cookie-compat/

**Comment by slandelle** [comment]
> The fact most user agents (receivers) are permitting zero-value Max-Age and treat it same as Expires set in the past, does not excuse server (sender) operating laxly.

When it comes to cookies, browsers first implement things, and then new behavior becomes official and turned into an RFC update. RFC6265 sadly lags behind, see for example how things happened for `SameSite`.

Your compatibility matrix shows forcefully expiring a cookie with a max-age <=0 is a feature supported in Safari 9 (current is 13), Chrome 49 (current is 81) and FireFox 44 (current is 75).
The bug you mentioned concludes that this feature works and the bug couldn't be reproduced.

In short, this is an intended feature and IMHO, it would be a bad idea to stop supporting it by default.

**Comment by normanmaurer** [comment]
@slandelle so just to ensure I did get it... you say all is good and we should just close this one ?

**Comment by slandelle** [comment]
@normanmaurer Yes

**Comment by normanmaurer** [comment]
ok lets close it then...
